### DISCUSSION TITLE
[주차별 리뷰] ※20210109 ~ 20210115 2주차 과제 리뷰※

### DISCUSSION THREAD

**Discussion by yoomambo** [4Moyede/HexatonClass-01#117] [01_주차별 리뷰]
여러분 안녕하세요.

멘토 유혁준, 강형구입니다.

여러분들의 2주차 과제를 다 확인했습니다.

## 전체

1. 과제 원본 파일 위에 그대로 사용하지 않은 학생들이 많습니다.

![data](https://user-images.githubusercontent.com/54018187/104611293-50901580-56c8-11eb-80f6-ef88eb36c2f3.png)

제대로 사용한 예시입니다. 다음 과제도 이렇게 해주시길 바랍니다.

2. 멘토의 피드백과 멘티들의 코드를 보고 본인의 과제를 리뷰하시기 바랍니다.

3. 추가적으로 issue 등록하라고 적어둔 부분은 수업 전 까지 issue 등록하시고 각자 댓글 한개씩 달아보시길 바랍니다.

4. 토요일 저녁 7시50분까지 줌으로 들어오시면 됩니다.

### 성준

- Issue
    - 지난주 issue 정리해서 마크다운 파일 저장 후, 과제폴더에 올릴 것
    - 추가적으로 해야할 것 : 다른 멘티들 과제들 보고 과제별 리뷰 2주차에 댓글로 review 해주기
- Python
    - 잘한 점
        - accommodates의 value 들 출력
            - sorted와 unique를 묶어서 잘 적용한 것 같네요! 다른 분들도 보고 배우면 좋을 것 같습니다.
        - 원하는 column만 가지는 DataFrame 만들기
            - pandas.DataFrame가 for문으로 iteration이 된다는 것을 너무 잘 활용한 것 같아요!
        - 선택한 column 의 value의 종류 파악
            - pandas.DataFrame에 set이 적용된다는 것을 알고 계셨네요! 매우 좋습니다
    - 보강해야할 점
        - 원하는 column만 가지는 DataFrame 만들기
            - 어쨌든 코드는 적었으니  코드르 좀 더 보완할 점, 그리고 성준 멘티님이 고민했던 부분을 issue에 등록하시길 바랍니다!
        - 선택한 column 의 value의 종류 파악
            - 각 column들의 값들이 지금 code에서 반복이 되는데 혹시 좀 더 간결하게, 효율적으로 바꿔보면 좋지않을까요?
        - Filtering해서 DataFrame 만들기
            - Filter들의 변수 이름을 어떻게 좀 직관적으로? 알아보기 쉽게 바꿀수 있을까요? issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
        - 성준 멘티님이 적은 코드를 함수화 시켜서 다시 바꿔보시길 바랍니다.

### 지우

- Notion
    - Project Proposal 올리기
- Issue
    - 지난주 issue 정리해서 마크다운 파일 저장 후, 과제폴더에 올릴 것
    - 추가적으로 해야할 것 : 다른 멘티들 과제들 보고 과제별 리뷰 2주차에 댓글로 review 해주기
- Python
    - 잘한 점
        1. Filtering해서 DataFrame 만들기
            1. filter들을 각각 변수들로 만들어서 지정을 한 것이 너무너무너무 좋았습니다 베리베리 굿!!! 이거 다른 멘티 분들도 본받을만합니다. 
        2. Price Filtering 한 data visualization
            1. histogram method를 유용하게 잘 사용하였네요. 굿굿
    - 보강해야할 점:
        1. 쥬피터파일 열어보면 image가 안보이는데, 어떤 이유일까요? issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
        2. csv file 불러와서 DataFrame 제작
            1. data라는 변수는 어떤 data인가요?
        3. accommodates의 value 들 출력
            1. `for i in ~~`  이 부분에 i를 쓰면 안됩니다. 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
            2. `print("This is accommodates'" + "s value : "` 부분에서 accomodate와 s를 왜 띄어서 +를 적용한 부분이 과연 필요할까 싶네요... 바꾸면 어떻게 될지, 왜 바꾸어야할지  issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
            3. `no_overlap_accommodates` 을 첫번째 줄에서도 쓰고 두번째 줄에서도 쓰면 왜 안좋을까요?  왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.

                ```python
                no_overlap_accommodates = no_overlap['accommodates']
                no_overlap_accommodates = no_overlap_accommodates.sort_values()
                ```

        4. 원하는 column만 가지는 DataFrame 만들기
            1. print가 너무 많은데... 어떻게 바꾸면 좋을까요? 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
        5. 선택한 column 의 value의 종류 파악
            1. print가 너무 많은데... ? 그리고 end를 써서 잘하긴 했는데, 뭔가 간결하게 해볼 수 있지 않을까요? 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
        6. 지우 멘티님이 적은 코드를 함수화 시켜서 다시 바꿔보시길 바랍니다.

### 경훈

- Notion
    - Proposal 올리기
- Issue
    - 지난주 issue 정리해서 마크다운 파일에 정리해서 과제폴더에 올릴 것
    - 추가적으로 해야할 것 : 다른 멘티들 과제들 보고 과제별 리뷰 2주차에 댓글로 review 해주기
- Python
    - 잘한 점 :
        1. accommodates의 value 들 출력
            1. drop_duplicates, sort_values 같은 메소드를 적절히 활용하셨네요! 이 부분도 칭찬합니다. 멘토도 처음본 메소드에요
            2. loc method 를 사용 잘 하셨네요!
        2. Price Filtering 한 data visualization
            1. histogram method를 적절하게 잘 사용한 것 같습니다.
    - 보강해야할 점:
        1. 과제하면 과제 파일 밑에다가 코드 작성해서 한 번에 제출해주세요 제발 플리즈~~ 우리가 문제를 한 번에 못봅니다..

            → 멘토 : 엉엉ㅠ

        2. 전체적으로 코드에 엔터가 없어서 알아보기 불편합니다. 코드를 읽는 다른사람이 된다고 가정하고 한 번 readable 하도록 바꿔보세요.
        3. 경훈 멘티님이 적은 코드를 함수화 시켜서 다시 바꿔보시길 바랍니다.
        4. accommodates의 value 들 출력
            1. for 문 안에 i가 무엇인가요? 협업자들이 보기에는 이해를 못하겠죠? 그리고 print 문을 좀 더 간결하게 바꿔보세요
        5. 선택한 column 의 value의 종류 파악 
            1. 출력물이 다르네요.. 다시해보시길 바랍니다.
        6. Price Filtering 한 data visualization 
            1. `import matplotlib.pyplot as plt` 같은 패키지를 import해서 선언하는 구문은 하단에 쓰면 안됩니다. 이유가 무엇일까요? issue에 등록해서 다같이 논의하길 바랍니다.

### 덕규

- Notion
    - Proposal 올리기
- Issue
    - 지난주 issue 정리한 후, 마크다운 파일 생성해서 자기 과제폴더에 올릴 것
    - 추가적으로 해야할 것 : 다른 멘티들 과제들 보고 과제별 리뷰 2주차에 댓글로 review 해주기
- Python
    - 잘한 점 :
        1. 전체적으로 노력한 부분이 보입니다! 구글링 잘 해서 사용하셨네요!
        2. 2번 : drop_duplicates, sort_values 같은 메소드를 적절히 활용하셨네요! 이 부분도 칭찬합니다. 멘토도 처음본 메소드에요
        3. 전체적으로 노력한 부분이 보입니다! 구글링 잘 해서 사용하셨네요!
        4. 2번 : drop_duplicates, sort_values 같은 메소드를 적절히 활용하셨네요! 이 부분도 칭찬합니다. 멘토도 처음본 메소드에요
    - 보강해야할 점:
        1. 지난 1주차 과제 피드백 적용 안한 것 같습니다! 확인하시고 수정해서 다시 pull request 부탁드립니다.
        2. 함수를 정의해서 풀지 않았네요... 덕규 멘티님이 적은 코드를 함수화 시켜서 다시 바꿔보시길 바랍니다.
        3. accommodates의 value 들 출력
            1. `import numpy as np` 같은 패키지 import 선언은 왜 하단에 하면 안될까요? 경훈 멘티님이 이슈 등록해주시면 issue에서 discussion 해보시길 바랍니다.
            2. `np.array(airb_new_sorted_by_values['accommodates'].tolist())` 이 부분 code 잘 돌아가는지 확인하려고 한 것 같은데, 그런 부분 있으면 미리 미리 지워주세요. 
            3. `list(np.array(airb_new_sorted_by_values['accommodates'].tolist()))` 이 부분을 그냥 적어놓으면 협업자들이 이해할까요?
            4. `for i in list` 이 부분에 i를 쓰면 안됩니다. 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
            5. 2번에 불필요한 코드들이 너무 많아요...
        4. 원하는 column만 가지는 DataFrame 만들기
            1. 만약 덕규님이 원하는 컬럼이 100개라면 다 적으실건가요? 메소드를 써서 간결하게 바꿔보세요
        5. 선택한 column 의 value의 종류 파악
            1. del 을 쓰는 습관은 안가지는것이 좋습니다. 좀더 간결하게 여러가지 함수나 메소드를 써서 해보시길 바랍니다.

                → 만약에 원래 테이블이 필요하다면? 변수란 사용할 데이터를 저장하고 사용하는 것입니다. 파일을 읽어온 소중한 데이터는 그대로 두고 새로운 변수에 필요한 것만 빼서 저장하는 느낌으로 하면 더 좋을 것 같습니다.

        6. Filtering해서 DataFrame 만들기
            1. 처음 코드에 `len(airb_filter.index)` 이 부분이 안돌아가는데 안돌아가는 이유가 무엇일까요? 주피터 노트북과 연관지어서 생각해보시길 바랍니다. 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.

                → Hint : Kermel - Restart & Run All를 해보시면 이해가 빠를 거에요!

        7. Price Filtering 한 data visualization
            1. list를 변수로 쓰면 안됩니다. 이유가 무엇일까요? 왜 그럴지 issue 등록해서 멘티들과 같이 논의 해주시길 바랍니다.
            2. 히스토그램 안에 제목, x축 label, y축 label이 없습니다. 추가해주세요.

### 재범

- Issue
    - 지난주 issue 정리해서 마크다운 파일에 저장 후, 자기 과제폴더에 올릴 것
    - 추가적으로 해야할 것 : 다른 멘티들 과제들 보고 과제별 리뷰 2주차에 댓글로 review 해주기
- Python
    - 리뷰 없음

**Comment by ghost** [comment]
> # 덕규
다시 수정하신 과제는 멘토분들이 피드백한거 다 수정하셔가지고 제 수준에서는 딱히 리뷰할만한 사항이 없네요.
주석으로 달아주신 부분에서 마치 함께 대화하면서 코드를 보는 듯한 생생함을 느꼈습니다. 굿굿
매일 파이썬 공부한다고 고생하시더니 노력은 배신하지 않나봅니다..ㅎㅎ

<br>

> # 지우

### 2번
```
no_overlap = data.drop_duplicates("accommodates", keep="first")
no_overlap_accommodates = no_overlap['accommodates']
no_overlap_accommodates = no_overlap_accommodates.sort_values()
for i in range(0,len(no_overlap_accommodates)) :
    print("This is accommodates'" + "s value : "  + str(no_overlap_accommodates.values[i]))
```
이 코드를 이렇게 변경할 수 있을 것 같아요.
```
for i in range(0,len(no_overlap_accommodates)) :
    print("{} : {}".format("This is accommodates\' s value" , no_overlap_accommodates.values[i]))
```

### 4번
출력 형식은 맞았지만 출력하는 코드가 좋은 코드 같지는 않습니다.  저도 고민하다가 결국 이상하게 내긴 했지만 더 깔끔한 방법 찾을 수 있게 고민해봐요~~


### 6번
```
fig = plt.figure()

ax = fig.add_subplot(1,1,1)
```
이 코드는 없어도 구동이 잘 되는데 혹시 확인해 보셨나요?

<br>

> # 경훈
일단 멘토님들이 피드백해주신대로 띄워쓰기를 하시면 가독성이 올라갈 것 같습니다.

```
print("We select only Good information about airbnb dataset : ")
print("1  :  room_type")
print("2  :  neighborhood")
print("3  :  reviews")
print("4  :  overall_satisfaction")
print("5  :  accommodates")
print("6  :  bedrooms")
print("7  :  price")
print("8  :  minstay")
```
이 코드 같은 경우에는
```
print('''We select only Good information about airbnb dataset :
          1  :  room_type
          2  :  neighborhood
          3  :  reviews
          4  :  overall_satisfaction
          5  :  accommodates
          6  :  bedrooms
          7  :  price
          8  :  minstay''')
```
이렇게 한번에 묶어서 쓸 수 있을 것 같아요.

<br>

> # 성준

### 2번
unique() 메소드를 사용하셨네요. 어디서 찾으셨나요... 좋은거 배우고 갑니다.

### 3번
Issue에 적으셨던 내용의 흔적들이 남아있네요. 그 부분 잘 추가하시면 좋은 발전 있을 것 같습니다.

### 4번
저는 이걸 방법을 못 찾아서 리스트로 비슷하게만 표현했는데 정확하게 구현하셨네요. 잘 배우겠습니다.

### 5번
Issue에 올리셨던 변수명에 대한 글을 잘 읽었습니다. 줄바꿈을 적극 활용해보시는것도 실력 발전에 도움이 될 것 같습니다.

### 6번
다른 분들 코드와 비교했을 때 딱 필요한 코드만 적으셨네요. 각 코드가 뭘 의미하는지 정확하게 인지하시고 계신 것 같습니다.

성준님은 항상 많은 시간 투자하여 고민하시고,, 또 그만큼 좋은 결과물을 내시는 것 같아서 과 선배로서 부끄럽기도 합니다. 그만큼 많은 도움 얻고 있으니 Issue 많이 올리셔서 고민거리 마구마구 던져주세요. 화이팅~~

<br>

> # 멘토님

멘토님 코드를 보고는 딱 두가지만 중점적으로 기억하고 싶네요.

### 1. enumarate 함수
```
* 반복문 사용 시 몇 번째 반복문인지 확인이 필요할 수 있습니다. 이때 사용합니다.
* 인덱스 번호와 컬렉션의 원소를 tuple형태로 반환합니다.
```
이렇게 두 가지 설명으로 정리할 수 있는데, tuple 형식으로의 반환이 꽤나 흥미로웠습니다.
```
for i, v in enumerate(t):
   print("index : {}, value: {}".format(i,v))
>>
index : 0, value: 1
index : 1, value: 5
index : 2, value: 7
index : 3, value: 33
index : 4, value: 39
index : 5, value: 52
```
이런 방식으로의 활용도 가능하다는 것을 배웠습니다.

### 2. 적극적인 사용자 정의 함수 사용

이 부분은 숙련도의 문제인 것 같은데, 저는 아직 함수를 정의하여 사용하는 것에 익숙치가 않아서 모든 코드를 날것 그대로 적으려는 습관이 있습니다. 그러다 보니 중복되는 코드도 많이 발생하여 나중에 다시 함수를 정의, 수정하는 방식으로 시간을 많이 잡아먹습니다. 

보다 적극적으로 함수 정의를 사용하여 가독성을 위해 노력해보겠습니다.

**Reply by yoomambo** [reply]
@JaebeomHan 덕규멘티님한테 리뷰한 것 처럼 격려의 메세지도 좋습니다! 굿굿

**Comment by lsj0822** [comment]
### 덕규 멘티님

3,4 : Issues에 올라온것을 적극적으로 활용하신것이 좋습니다! 저도 이슈에 좋은 방법들이 올라오면 과제에 직접 활용해봐야겠어요

5 : 결과의 개수를 세는 코드를 len()함수를 이용하여 만들었네요! 이 부분을 어떻게 구현할지 고민이 많았는데 len함수가 정말 쓰임이 무궁무진하다는 것을 느끼고 갑니다...

그 외에 주석을 이용하여 과제를 풀때 고민한 흔적이 보여서 이 부분이 정말 좋았습니다.

### 경훈 멘티님

5 : 필터링을 할때, loc함수를 이용하여 행에 조건을 걸으신 부분이 인상적이었습니다! 저 포함 다른 멘티분들은 loc이나 iloc함수를 사용할때 행은 그대로 놓고 열에 제약을 줬는데, 반대로 행에다가 제약을 준다는 생각을 못했는데 loc을 다양하게 활용하는 게 좋았어요. 나중에 pandas를 사용하면 이렇게도 써봐야겠어요


### 재범 멘티님

2,3 : format을 활용하여 print를 한 부분이 인상적이었습니다! format을 활용하니까 되게 깔끔한 코드가 나오네요. 

그리고 코드에서 줄바꿈을 적극적으로 활용한 부분이 인상적이었습니다. 가독성이 높아서 코드를 리뷰할때 정말 쉽게 읽었습니다! 이슈에서 말해주신대로 줄바꿈만 잘해도 훨씬 깔끔한 코드가 나오는 것 같아요. 앞으로 자주 써먹어야겠네요

### 지우 멘티님

4 : 4번은 원래 딕셔너리를 이용하여 출력하는건데, 이를 직접 출력하려고 코드를 짜신 부분이 인상적이었습니다. 물론 이 방법이 의도한 결과는 아니었지만, 수고 많으셨어요... 

5 : all_filtering 부분이 되게 깔끔하네요! 저같은 경우 data['price'] > 250같은 조건들을 필러팅 변수에 전부 넣는 방식으로 코딩했는데, 이렇게 안하고 변수로 바꾼다음 이를 대입하니까 훨씬 보기 편한 코드가 됐어요.


### 멘토님 코드

2 : set를 4번뿐만 아니라 여기에서도 활용할 수 있구나... 집합 자료형이 얼핏봤을때는 아예 안쓰거나 드물게 쓸 줄 알았는데, 집합 자료형이 활용범위가 무궁무진하네요

3 : 코드에서 column을 더하는 방식이 아니라 오히려 빼는 방식인게 되게 좋았습니다! 또한 enumerate라는 함수가 지금까지 코딩하면서 필요한 함수였는데(그 전에는 index라는 변수를 for문에서 1개씩 더하는 방식으로 코딩) 되게 좋은 함수고, 앞으로 자주 활용하게 될 함수인것 같아요. 코드가 좀더 깔끔해질것같아요

4 : column_list 라는 변수를 이용하여 이를 for문에 넣어서 한꺼번에 딕셔너리와 집합 자료형으로 표현할 수 있네요! 

5 : 이 역시 인자를 3개 가지고 있는 함수를 만들어서 표현한 부분이 정말 좋아요! 앞으로 함수를 만들어서 써야할일이 많을텐데, 이런식으로 간단한 코드라도 함수화하는 연습을 꾸준히 해야겠다는 생각이 드네요.

**Comment by jiwoo4424** [comment]
**성준**

3. 필요한 column의 인덱스 번호를 리스트 형식으로 추출함으로써 좀 더 깔끔한 코드가 된 거 같아요 ! 다음에 한 번 써봐야겠습니다

4. 문제에는 room_type, neighborhood 들의 value들을 한 dictionary에 넣어서 출력하라고 적혀있는데 reviews 가 추가된 거 같아요 !!

5. input을 활용해서 조건을 입력받도록 한 부분을 보면서 성준님이 코드들은 꼼꼼하고 편리하게 만들려고 노력하는 ? 느낌이 들었습니다 !

6. color='#ffb2b2' 이 부분을 어떻게 구글링해서 찾아냈는지 신기합니다 ,,,

**재범**

3. 첫번째 코드는 저도 똑같이 짰는데 저는 그 뒤 코드들을 아래와 같이 한 줄로 적어서 좀 더 코드를 간단하게 만들고자 해보았습니다 ! 
no_overlap_accommodates = no_overlap['accommodates'].sort_values()

4. format을 이용해 print 구문을 만드셨는데 책으로 접한 format을 생생한 예로 보는 느낌 ,,,, 너무 달아

5. set을 만드는 함수를 구현해서 print하신 게 인상깊어요 ! 코드들에 전체적으로 줄바꿈이 잘 되어 있어서 가독성이 너무 좋은 코드들이네용 ㅎㅎ

6. 재범님의 코드에는 항상 len이 빠지지 않고 등장하는 거 같아요 ! 간단한 코드인 len에 대해서 다시 생각해보게 됩니다.....

**경훈**

3. 시리즈를 리스트로 변환하는 accommodates_df.values.tolist() 요 코드가 몰랐던 부분이여서 기억해뒀다가 써먹어 보겠습니당

4,5,6 . 코드에 모두 함수를 이용해서 푼 것을 보고 함수 이용을 정말 좋아하시고 잘하시는구나,,,, 생각이 들면서 부럽습니다 !

5. df_filtering = df.loc[filtering, :] 이 부분이 정확히 뭔지 몰라서 구글링해서 찾아보았는데 오우... 너무 신선한 코드입니다 !!!

**덕규**

3,4. airb.iloc[:, [2,4,5,6,7,8,9,10]] 등등의 코드를 보면 iloc 활용을 잘하셔서 깔끔하게 코딩이 되어 있는 느낌을 받았습니다. 

전체적으로 코드가 너무 간결하고 깔끔해서 정말 딱히 적을 말이 없네요 !! 파이썬 실력이 많이 좋아진 게 딱 보이는 거 같아요 ! 노력하시는 모습이 자극됩니다 ~~~~~ 같이 열심히 해보자구요

**Reply by lsj0822** [reply]
@jiwoo4424 4번에서 #ffb2b2 같은거는 컬러코드로 pandas뿐만 아니라 웬만한 색깔을 나타낼 수 있는 프로그램에서는 쓸 수 있는 코드에요! 구글이나 네이버에 rgb to hex라고 치면 컬러코드가 나와요

**Comment by NA-DK** [comment]
### 성준
 1.  unique 함수는 구글링했을 때 못본 함수 같은데 duplicate를 쓰는방법 이외에도 하나 더 있었군요...! 하나 알고 갑니다.
 
 2.  list_num을 정의해서 깔끔한 코드를 쓴 것도 좋은 것 같습니다.
 
### 경훈
1. import 구문을 코드 제일 처음에 써줌으로써 뒤에 나올 내용을 쉽게 예측해볼 수 있는 효과를 갖는다는 걸 알게되었습니다.  습관처럼 사용해야 할 것 같아요.

2. 저는 Histogram의 색상을 선택할 때 color='#ffb2b2'로 했는데 'r'이라고 지정해도 적용이 되는군요...! 저렇게 쉬운 방법이..

### 재범
1. 처음 csv파일을 불러올 때 상대경로를 한번 사용해보았으면 좋겠다는 생각을 해보았습니다...!

2. print("{} : {}".format("This is accommodates", pick_column.values[value])) 
이 부분 신기한데요..? 다시 한번 찬찬히 읽으면서 이해했습니다..! format을 사용하는 것도 익혀봐야겠어요.

3. 자료형 중 튜플형식을 채택한 게 인상깊었습니다. 다양한 자료형을 적용해보는걸 저도 해보고 싶습니다. 

### 지우

1. 전반적으로 변수명이나 함수명을 정말 잘 정하는것 같아요 가독성이 뛰어난 코드인 것 같습니다. 

2. 늘 말하지만 깔끔한 코드를 구현하는 거 같아서, 코드를 처음에 어떻게 짜고 적어내려가기 시작하는지 그 과정이 궁금합니다..!

**Comment by dlrudgnsdlek** [comment]
리뷰가 늦었습니다. 죄송합니다.

지우
-----
- 4번 문제에서 어떻게든 풀어내는 모습이 보기 좋았습니다!
- <pre><code> for i in range(len())</code></pre> 을 사용하신 덕분에 issue를 통해  enumerate에 대해 깔끔하게 정리가 되었습니다!

덕규
----
- iloc  메소드를 이용해 열번호만 적시해도 적시된 열이 추출되는 방식으로 원하는 Column만 뽑아내는게 인상 깊었습니다!
- 히스토그램에서 fontsize로 글씨 크기까지 생각하신게 인상적이었습니다:)
- 결과의 수를 세는 데 len() 함수를 이용한것도 정말 잘 한것 같아요! 여러모로 전혀 생각하지 못한 코드들을 가장 많이 본 것 같습니다.

성준
-----
- 성준 멘티님의 코드를 보고 unique 함수와 set 함수에 대해 다시 공부 해 보았습니다.. 감사합니다! 
- 필요한 column의 인덱스 번호를 리스트로 뽑아낸다는 처음 생각을 정말정말정말 잘 하신 것 같아여

재범
-----
- 아래 코드에서
<pre><code>for value in range(0,len(pick_column)):
    print("{} : {}".format("This is accommodates", pick_column.values[value]))</code></pre> 
format을 이용하면 이렇게 간단하다는 것을 배웠습니다! 저의 코드가 많이 반성이 되네요..
- rwidth 를 통해 히스토그램의 막대 간격까지 설정한게 신기했습니다.
